homeassistant/components/thethingsnetwork/config_flow.py

```diff
@@ -0,0 +1,117 @@
+"""The Things Network's integration config flow."""
+
+from collections.abc import Mapping
+import logging
+import traceback
+from typing import Any
+
+from ttn_client import TTNAuthError, TTNClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, ConfigFlowResult
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import CONF_API_KEY, CONF_APP_ID, CONF_HOSTNAME, DOMAIN, TTN_API_HOSTNAME
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TTNFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize flow."""
+        self._reauth_entry: ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: Mapping[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """User initiated config flow."""
+
+        errors = {}
+        if user_input is not None:
+            connection_error = None
+            try:
+                client = TTNClient(
+                    user_input[CONF_HOSTNAME],
+                    user_input[CONF_APP_ID],
+                    user_input[CONF_API_KEY],
+                    0,
+                )
+                await client.fetch_data()
+            except TTNAuthError:
+                # Raising ConfigEntryAuthFailed will cancel future updates
+                # and start a config flow with SOURCE_REAUTH (async_step_reauth)
```

Comment:
Not relevant here

Reply:
The comment, this comment is talking about raising an exception, even tho that's not relevant here

Reply:
indeed - old comment from the old code - removed as the code is clear now anyway